### DISCUSSION TITLE
compiler_depend.make incorrect paths

### DISCUSSION THREAD

**Discussion by bobrots** [msys2/MSYS2-packages#3015] [Q&A]
I'm not certain if I should go here to ask questions but I've noticed something with the CMake package that CMake for windows doesn't do. The file `compiler_depends.make` produces wrong paths for all dependencies. I'm not familiar with CMake so I might make a mistake somewhere, however, when CMake for windows is used it actually produces correct relative paths like `../STM32Core/Core/Src/adc.c`. However, for the cmake package installed through MSYS2 it produces `STM32Core/C/:/Users/UserName/Documents/repos/Flumbler/src/STM32Core/Core/Src/adc.c` which is a completely invalid path. 

Could anyone give me some suggestions on what is going on here? Again, CMake for windows doesn't do this. 

Snippet of compiler_depend.make:

`STM32Core/CMakeFiles/STM32Core.dir/Core/Src/adc.c.obj: ../STM32Core/Core/Src/adc.c \
  STM32Core/C/:/Users/UserName/Documents/repos/Flumbler/src/STM32Core/Core/Src/adc.c \
  STM32Core/C/:/Users/UserName/Documents/repos/Flumbler/src/STM32Core/Core/Inc/adc.h \
  STM32Core/C/:/Users/UserName/Documents/repos/Flumbler/src/STM32Core/Core/Inc/main.h \`

**Comment by Biswa96** [comment]
Please provide the details about the project and steps to reproduce your issue.

**Comment by bobrots** [comment]
@Biswa96 : I've generated a makefile project with STM32CubeMx. After that I modified it to use CMake to be able to build (as best as I could) and created an object library out of it and added another object library to function as the project code in C++. The project is intended to work in VS Code. The project is included in the attachment.

@MehdiChinoune : If I'm not mistaken I'm using cmake from MSYS (`which cmake` on windows gives me `/usr/bin/cmake`. And the generator I'm using is "Unix Makefiles"

[CMakeTest.zip](https://github.com/msys2/MSYS2-packages/files/8756524/CMakeTest.zip)
.

For your information, a full list of the packages I'm using:

- mingw-w64-x86_64-arm-none-eabi-gcc
- mingw-w64-x86_64-gdb-multiarch (for debugging)
- mingw-w64-x86_64-openocd
- cmake
- make

EDIT: Added missing package.

**Comment by elupus** [comment]
Just running into same issue. It could maybe depend on the compiler used. A mingw compiler will output windows paths for deps, which might confuse a msys cmake.

**Reply by elupus** [reply]
I can add that if CMAKE_DEPENDS_USE_COMPILER (https://cmake.org/cmake/help/latest/variable/CMAKE_DEPENDS_USE_COMPILER.html) is disabled. This issue does not occur. That flag happens to be the default on newer cmake.

**Reply by elupus** [reply]
I've opened a bug upstream: https://gitlab.kitware.com/cmake/cmake/-/issues/24228
